### PR TITLE
Make ruff configuration more strict

### PR DESCRIPTION
**Pull request by orsinium**
1. Enable all ruff rules that don't cause any trouble in the current code.
2. Resolve the reported violations.

### CODE REVIEW

**Comment by orsinium** [comment]
Damn, lintje failed.

**Comment by tombruijn** [comment]
> Damn, lintje failed. Something about changeset?

The changeset one is a hint, but for this change it's not necessary. We use it to build our changelogs.

The error is that you made a commit without starting the subject with an uppercase letter. The rests are just hints.

File: src/appsignal/config.py [hunk 1]
```diff
@@ -116,11 +125,11 @@ def option(self, option: str) -> Any:
         return self.options.get(option)
 
     @classmethod
-    def load_from_system(_cls) -> Options:
```
Comment (tombruijn) [inline src/appsignal/config.py:119]: We can't prefix it with an underscore, even if we don't use it?

**Review by tombruijn — APPROVED** [review]

File: src/appsignal/config.py (hunk 1 above)
Comment (orsinium) [reply src/appsignal/config.py:119]: Yes. The convention is to always use `cls` for the first argument of classmethods. If you don't need the class in the method, use `@staticmethod` instead.

Comment (unflxw) [reply src/appsignal/config.py:119]: Ah, then these should either be in the `Options` object (and use `cls` to instantiate the class) or be modified to use `@staticmethod`. We can change that in a separate PR.

**Review by unflxw — APPROVED** [review]

File: src/appsignal/config.py (hunk 1 above)
Comment (tombruijn) [reply src/appsignal/config.py:119]: Yes, let's make it a static method then. It doesn't need to know the class's state.

**Pull request merged**